doc-content/drools-docs/src/main/asciidoc/AuthoringAssets/TestScenarioEditor-section.adoc

```diff
@@ -1,75 +1,128 @@
+image::project-data/test-scenario-expected-rules.png[Add EXPECT results to the test scenario]
++
+The list includes the following options, depending on the data in the *GIVEN* section and the data objects available in the *Data Objects* tab of the test scenarios designer:
+
+* *Rule:* Use this to specify a particular rule in the project that is expected to be activated as a result of the *GIVEN* input. Type the name of a rule that is expected to be activated or select it from the list of rules, and then in the test scenarios designer, specify the number of times the rule should be activated.
+* *Fact value:* Use this to select a fact and define values for it that are expected to be valid as a result of the facts defined in the *GIVEN* section. The facts are listed by the *Fact name* previously defined for the *GIVEN* input.
+* *Any fact that matches:* Use this to validate that at least one fact with the specified values exists as a result of the *GIVEN* input.
++
+. Choose a fact for the desired expectation (such as *Fact value:* `application`) and click *Add* or *OK*.
+. Click the fact in the test scenarios designer and select the field to be added and modified.
++
+.Modify a fact field
+image::project-data/test-scenario-field-value.png[Modify a fact field]
++
+. Set the field values to what is expected to be valid as a result of the *GIVEN* input (such as `approved` | `equals` | `false`).
+. Continue adding any other *EXPECT* input data for the scenario and click *Save* in the test scenarios designer to save your work.
+. After you have defined and saved all *GIVEN*, *EXPECT*, and other data for the scenario, click *Run scenario* in the upper-right corner to run this `.scenario` file, or click *Run all scenarios* to run all saved `.scenario` files in the project package (if there are multiple). Although the *Run scenario* option does not require the individual `.scenario` file to be saved, the *Run all scenarios* option does require all `.scenario` files to be saved.
++
+If the test fails, address any problems described in the *Reporting* message at the bottom of the window, review all components in the scenario, and try again to validate the scenario until the scenario passes.
++
+. Click *Save* in the test scenarios designer to save your work after all changes are complete.
+
+=== Adding GIVEN facts in test scenarios
+
+The *GIVEN* section defines input facts for the test. For example, if an `Underage` rule in the project declines loan applications for applicants under the age of 21, then the *GIVEN* facts in the test scenario could be `Applicant` with `age` set to some integer less than 21.
+
+.Prerequisite
+All data objects required for your test scenario have been created or imported and are listed in the *Data Objects* tab of the test scenarios designer.
+
+.Procedure
+. In the test scenarios designer, click *GIVEN* to open the *New input* window with the available facts.
++
+.Add GIVEN input to the test scenario
```

Comment:
Would it not make more sense to document the "Add GIVEN..." _before_ "Add EXPECTED..." section?

Reply:
Thanks! I somehow switched these in this PR (when I copied and pasted). It's correct elsewhere. Fixed now. Sorry about that!